extensions/wikia/Email/Controller/FounderDigestController.class.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+use Email\Check;
+use Email\Tracking\TrackingCategories;
+
+abstract class FounderDigestController extends EmailController {
+	// Defaults; will be overridden in subclasses
+	const TRACKING_CATEGORY_EN = TrackingCategories::DEFAULT_CATEGORY;
+	const TRACKING_CATEGORY_INT = TrackingCategories::DEFAULT_CATEGORY;
+
+	protected $wikiId;
+	protected $pageViews;
+
+	public function initEmail() {
+		$this->wikiId = $this->request->getVal( 'wikiId' );
+		$this->pageViews = $this->request->getVal( 'pageViews' );
+		$this->assertValidParams();
+
+		$this->pageViews = number_format( $this->pageViews );
+	}
+
+	protected function getSummary() {
+		return $this->getMessage( 'emailext-founder-digest-summary',
+			\WikiMap::getWikiName( $this->wikiId ) )->parse();
+	}
+
+	protected function assertValidParams() {
+		$this->assertValidWikiId();
+		$this->assertPageViewsSet();
+	}
+
+	/**
+	 * Asserts that the wikiId parameter is valid
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertValidWikiId() {
+		if ( $this->wikiId < 1 ) {
+			throw new Check( 'Invalid value passed for `wikiId`' );
+		}
+	}
+
+	/**
+	 * Asserts that the `pageViews` value is passed
+	 *
+	 * @throws \Email\Check
+	 */
+	protected function assertPageViewsSet() {
+		if ( empty( $this->pageViews ) && $this->pageViews !== '0' ) {
```

Comment:
Yes but '0' is a valid value for page views here. That's why I'm checking that it's not '0'.